pkg/daemon/containernetwork/containernetwork.go

```diff
@@ -185,12 +184,10 @@ func ConfigureContainerNic(containerNicName, hostNicName, nodeIfName string, all
 	}
 
 	if allocatedIPs[networkingv1.IPv4] != nil {
-
-		ipv4AddressAllocated = true
 		// ipv4 address
 		defaultRouteNets = append(defaultRouteNets, &types.Route{
 			Dst: net.IPNet{IP: net.ParseIP("0.0.0.0").To4(), Mask: net.CIDRMask(0, 32)},
-			GW:  allocatedIPs[networkingv1.IPv4].Gw,
+			GW:  net.ParseIP(PodVirtualV4DefaultGateway),
```

Comment:
vlan and vxlan all use virtual gateway now?

Reply:
For vxlan, it's not needed either.